content/blog/2024/06/kubewarden-1-13-release.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Kubewarden 1.13 release
+authors:
+  - Flavio Castelli
+date: 2024-06-05
+---
+
+I'm pleased to announce a new release of Kubewarden, version 1.13. This release features a series of
+improvements and bug fixes that contribute to better performance and stability.
+
+Let's go through the most significant changes.
+
+## Policy Server memory usage
+
+A community member reported that the Kubewarden Policy Server was using a lot of memory, especially
+when running context aware policies on big clusters. The number of resources being accessed by the
+policies was significantly high, in the order of 3200 Namespaces, 10500 Ingresses, 200 ClusterRoleBindings
+and 11000 RoleBindings.
+Under certain circumstances, the amount of memory being used caused the Policy Server to be terminated by the kernel's OOM killer.
+
+After some investigation, we found two different root causes of this issue.
+
+The first one was related to [kube-rs](https://github.com/kube-rs), the Rust library we use to interact with Kubernetes.
+The library was doing excessive allocations when building the initial state of a resource. We fixed this issue
+with [this PR](https://github.com/kube-rs/kube/pull/1494).
+With this fix we reduced the peak of memory usage caused by the creation of the initial state of a Kubernetes resource.
+
+The second issue was related with how the memory allocator of Rust works. By default, Rust uses the system allocator (glibc malloc in our case).
+It turns out the system allocator is not releasing the allocated memory back to the system when this is no longer needed.
+We changed Policy Server to use the [jemalloc](http://jemalloc.net/) allocator which, on top of providing great performances, it has many
+configuration knobs.
+After some experimentations, we found the right balance between performances and memory usage.
+
+With these two changes, we were able to reduce the peak of memory consumption of the Policy Server significantly, plus we reduced the
+idle memory consumption.
+As a pleasant side effect, we also noticed an improvement in the performances of the Policy Server thanks to the usage of jemalloc.
+
+These are some numbers we obtained while doing performance tests of a Gatekeeper policy fetching, among other resources, 10000 RoleBindings:
+
+|                    | HTTP Request Duration (avg) | Max RSS Under load | Idle RSS |
+| ------------------ | :-------------------------: | :----------------: | :------: |
+| Policy Server 1.12 |          436.15ms           |       1.4 Gb       |  1.2 Gb  |
+| Policy Server 1.13 |          233.663ms          |       1.2 Gb       |  264 Mb  |
+
```

Comment:
This table looks small in my screen:

![image](https://github.com/kubewarden/kubewarden.io/assets/1514798/d63e1d2f-ad0e-4f45-9b44-b98501c78bc3)

I do not know if it is possible to make it bigger. If it is, we should do it.

Reply:
@jordojordo can you help me fix the CSS the proper way?

I've looked a bit into that and I **think** I found the issues. The `<p>` elements of the blog have `font-size: 2.1rem;` and `line-height: 2.8rem;`, while the contents of the table have "vanilla" settings, that leads to the table's text being smaller that the rest of the article

This is how the table looks like once I change these values

![image](https://github.com/kubewarden/kubewarden.io/assets/22728/b89fad3f-0bf2-45d0-bcb8-d560734429ca)

I don't know however how to properly change that. I don't want to hardcode these values, shouldn't we instead inherit them somehow?

Reply:
To me, it should go into `assets/sass/blog-main.scss` IIUC.

Reply:
I've fixed it with abc8fe7e6960b24265d2d21583f8d36f7d02bfa4

@jordon: please take a look at that once you're back